diff -Naur v5/src/content.js new-v5/src/content.js
--- v5/src/content.js	2018-07-12 17:54:48.521576015 +0200
+++ new-v5/src/content.js	2018-07-12 17:58:28.645575175 +0200
@@ -1,7 +1,7 @@
 
 import { Record, List, Set } from 'immutable'
 
-import * as Arbitrary from 'test/arbitrary'
+import * as Arbitrary from './test/arbitrary'
 
 
 const Content = Record({
diff -Naur v5/src/file-system.js new-v5/src/file-system.js
--- v5/src/file-system.js	2018-07-12 17:54:48.521576015 +0200
+++ new-v5/src/file-system.js	2018-07-12 18:07:36.817573084 +0200
@@ -1,13 +1,13 @@
 
 import { Map, List, Record } from 'immutable'
-import { generateRandomString } from 'random-gen'
-import duck from 'reducers/duck'
+import { generateRandomString } from './random-gen'
+// import duck from './reducers/duck'
 
-import * as Cache from 'cache'
-import * as Content from 'content'
-import * as Arbitrary from 'test/arbitrary'
-import * as Loop from 'test/loop'
-import tT from 'table-tree'
+import * as Cache from './cache'
+import * as Content from './content'
+import * as Arbitrary from './test/arbitrary'
+import * as Loop from './test/loop'
+import tT from './table-tree'
 const TT = tT(Content)
 
 
@@ -145,6 +145,12 @@
   return state
 }
 
+
+
+export const toJs = (state) => {
+  return state.toJS()
+}
+
 export const toStrList2 = Cache.make((state) => {
   return TT.toStrList2(state.get('tree'))
 })
diff -Naur v5/src/table-tree.js new-v5/src/table-tree.js
--- v5/src/table-tree.js	2018-07-12 17:54:48.521576015 +0200
+++ new-v5/src/table-tree.js	2018-07-12 17:58:12.377575237 +0200
@@ -1,9 +1,9 @@
 
-import * as Loop from 'test/loop'
-import * as Arbitrary from 'test/arbitrary'
-import * as Cache from 'cache'
+import * as Loop from './test/loop'
+import * as Arbitrary from './test/arbitrary'
+import * as Cache from './cache'
 
-import { generateRandomString } from 'random-gen'
+import { generateRandomString } from './random-gen'
 
 import { Map, Record, List } from 'immutable'
 
diff -Naur v5/src/test/arbitrary.js new-v5/src/test/arbitrary.js
--- v5/src/test/arbitrary.js	2018-07-12 17:54:48.521576015 +0200
+++ new-v5/src/test/arbitrary.js	2018-07-12 17:58:20.101575208 +0200
@@ -1,7 +1,7 @@
 
 import { Map, List } from 'immutable'
 
-import { generateRandomString } from 'random-gen'
+import { generateRandomString } from '../random-gen'
 
 import colors from 'colors/safe'
 
